variables.tf

```diff
@@ -434,6 +446,10 @@ variable "custom_origins" {
       origin_keepalive_timeout = number
       origin_read_timeout      = number
     })
+    origin_shield = object({
+      enabled = bool
+      region  = string
+    })
```

Comment:
It sucks that we have to make such a complex object even more complex. It's a breaking change, but there really isn't any other logical place for it IMO.

If we didn't have to opt in it'd be cool (https://www.terraform.io/language/functions/defaults)

But it is what it is.

Reply:
I agree. Let's hope the `defaults` function will soon move out of the experimental phase.

Reply:
@syphernl how much of a blocker is this for you? We're trying to keep breaking changes to a minimum, and yes, desperate for `defaults` function to graduate from experimental.

Reply:
@osterman Our customers would like to have this implemented in their environment. 
For the time being we could probably use our own fork/branch instead although from update-perspective that is not ideal.